packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -526,74 +610,236 @@ export const Tabs = (props: TabsProps) => {
+        role='tabpanel'
+        id={value}
+        ref={ref}
+        {...rest}
+      >
+        {children}
+      </Box>
+    );
+  } else {
+    return <></>;
+  }
+}) as PolymorphicForwardRefComponent<'div', TabsPanelOwnProps>;
+TabsPanel.displayName = 'Tabs.Panel';
+
+// ----------------------------------------------------------------------------
+// Tabs legacy component
+
+type TabsLegacyProps = {
+  /**
+   * Content displayed to the right/bottom of the horizontal/vertical tabs
+   *
+   * If `type = 'pill'`, `actions` is not applicable.
+   */
+  actions?: React.ReactNode[];
+  /**
+   * Elements shown for each tab.
+   * Recommended to pass an array of `Tab` components.
+   */
+  labels: React.ReactNode[];
+  /**
+   * Handler for activating a tab.
+   */
+  onTabSelected?: (index: number) => void;
+  /**
+   * Index of the active tab.
+   */
+  activeIndex?: number;
+  /**
+   * Control whether focusing tabs (using arrow keys) should automatically select them.
+   * Use 'manual' if tab panel content is not preloaded.
+   * @default 'auto'
+   */
+  focusActivationMode?: 'auto' | 'manual';
+  /**
+   * Color of the bar on the active tab.
+   * @default 'blue'
+   */
+  color?: 'blue' | 'green';
+  /**
+   * Custom CSS class name for tabs.
+   */
+  tabsClassName?: string;
+  /**
+   * Custom CSS class name for tab panel.
+   */
+  contentClassName?: string;
+  /**
+   * Custom CSS class name for the tabs wrapper.
+   */
+  wrapperClassName?: string;
+  /**
+   * Content inside the tab panel.
+   */
+  children?: React.ReactNode;
+} & TabsOrientationProps &
+  TabsOverflowProps;
+
+const TabsComponent = React.forwardRef((props, forwardedRef) => {
+  let actions: Array<React.ReactNode> | undefined;
+  if (props.type !== 'pill' && props.actions) {
+    actions = props.actions;
+    props = { ...props };
+    delete props.actions;
+  }
+
+  const {
+    labels,
+    onTabSelected,
+    focusActivationMode,
+    color,
+    activeIndex,
+    tabsClassName,
+    contentClassName,
+    wrapperClassName,
+    children,
+    ...rest
+  } = props;
+
+  const [currentActiveValue, setCurrentActiveValue] = React.useState<string>();
+  return (
+    <TabsWrapper className={wrapperClassName} {...rest}>
+      <TabList
+        color={color}
+        className={tabsClassName}
+        focusActivationMode={focusActivationMode}
+        activeValue={currentActiveValue}
+        setActiveValue={(value) => setCurrentActiveValue(value)}
+        ref={forwardedRef}
+      >
+        {labels.map((label, index) => {
+          const tabIndex = `label-${index}`;
+          return React.isValidElement(label) ? (
+            React.cloneElement(label as JSX.Element, {
+              value: tabIndex,
+              active: activeIndex === index || tabIndex === currentActiveValue,
+              onClick: (args: unknown) => {
+                onTabSelected?.(index);
+                setCurrentActiveValue(tabIndex);
```

Comment:
i don't think this `onClick` should be needed. it should be automatically handled in Tab. i believe the `onActivated` prop is what's missing

Reply:
I believe this was fixed during pair programming session.